### DISCUSSION TITLE
Multiple Databases for ODBC adapter?

### DISCUSSION THREAD

**Discussion by schlich** [tconbeer/harlequin#415] [Ideas (Feature Requests)]
Hello! what a wonderful tool.  I was wondering if you had plans on supporting multi-db servers in the Data Catalog pane? Usually (i.e. for SSMS/SQL Server) I see all of my server's databases listed-- maybe I am missing something, but I think I've looked through all the docs and help menus!

**Comment by tconbeer** [comment]
To be honest, I don't know how to enable this! I've tried using both pyodbc's [`cursor.tables()`](https://github.com/mkleehammer/pyodbc/wiki/Cursor#tablestablenone-catalognone-schemanone-tabletypenone) and querying `information_schema.schemata`, and both only return info on the currently-connected database (if there is a database in the connection string, or the "master" database otherwise) (at least on SQL Server on Azure).

If anybody knows how to query for the full list of all databases, I'd be grateful.

**Reply by schlich** [reply]
[Stack Overflow](https://stackoverflow.com/questions/147659/get-list-of-databases-from-sql-server) seems to suggest 

```sql
SELECT name FROM master.dbo.sysdatabases
```

Is that relevant?

**Reply by tconbeer** [reply]
Yes!

**Reply by tconbeer** [reply]
of course this is sql-server specific; oracle and others don't have this. But we could try it and follow that path if the query succeeds. We would then have to iterate over the list of returned dbs and make [multiple calls](https://stackoverflow.com/questions/2875768/how-do-i-list-all-tables-in-all-databases-in-sql-server-in-a-single-result-set) to `<db>.information_schema.tables` . Not that hard, just annoying that there isn't an odbc-standard way of doing this.

**Reply by schlich** [reply]
I'm certainly down to put in a PR for this "when some time frees up" 😉 (but actually)

**Reply by tconbeer** [reply]
That would be much appreciated

**Comment by tconbeer** [comment]
This has been fixed in harlequin-odbc v0.3.0: https://github.com/tconbeer/harlequin-odbc/releases/tag/v0.3.0
